File: 2712-minimum-cost-to-make-all-characters-equal/2712-minimum-cost-to-make-all-characters-equal.cpp
class Solution {
public:
    long long minimumCost(string s) {
        long long ans=0;
        for(int i = 0;i+1<s.size();i++){
            if(s[i]!=s[i+1]){
                ans+=min(i+1,(int)s.size()-i-1);
            }
        }
        return ans;
    }
};

File: 2712-minimum-cost-to-make-all-characters-equal/2712-minimum-cost-to-make-all-characters-equal.py
class Solution(object):
    def minimumCost(self, s):
        """
        :type s: str
        :rtype: int
        """
        a = 0
        for i in range(1, len(s)):
            a += 0 if s[i - 1] == s[i] else min(i, len(s) - i)
        return a